homeassistant/components/zwave_js/api.py

```diff
@@ -280,6 +291,53 @@ def node_removed(event: dict) -> None:
     )
 
 
+@websocket_api.require_admin  # type:ignore
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/set_config_parameter",
+        vol.Required(ENTRY_ID): str,
+        vol.Required(NODE_ID): int,
+        vol.Required(PROPERTY): int,
+        vol.Optional(PROPERTY_KEY): int,
+        vol.Required(VALUE): int,
+    }
+)
+async def websocket_set_config_parameter(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict
+) -> None:
+    """Set a config parameter value for a Z-Wave node."""
+    entry_id = msg[ENTRY_ID]
+    node_id = msg[NODE_ID]
+    property_ = msg[PROPERTY]
+    property_key = msg.get(PROPERTY_KEY)
+    value = msg[VALUE]
+    client = hass.data[DOMAIN][entry_id][DATA_CLIENT]
+    node = client.driver.controller.nodes[node_id]
+    try:
+        result = await async_set_config_parameter(
+            node, value, property_, property_key=property_key
+        )
+    except (InvalidNewValue, NotFoundError, NotImplementedError, SetValueFailed) as err:
+        code = ERR_UNKNOWN_ERROR
+        if isinstance(err, NotFoundError):
+            code = ERR_NOT_FOUND
+        elif isinstance(err, (InvalidNewValue, NotImplementedError)):
+            code = ERR_NOT_SUPPORTED
+
+        connection.send_error(
+            msg[ID],
+            code,
+            f"{type(err).__name__}: {','.join(err.args)}",
```

Comment:
How will this error message look? Why do we need to join the exception arguments? Are there more than the message?

Reply:
as an example, when we get `SetValueFailed`:

`SetValueFailed: Unable to set value, refer to https://zwave-js.github.io/node-zwave-js/#/api/node?id=setvalue for 
 possible reasons`

As of now we only have one args so I could hardcode `err.args[0]`

Reply:
Or just pass the exception instance. It has a magic method `__str__`.

Reply:
I'm not sure what I am doing wrong, but I tried with and without `str()` for the following:
`f"{str(err)}, {type(err).__name__}: {','.join(err.args)}"`

```python
DEBUG    homeassistant.components.websocket_api.http.connection:http.py:76 [4469081664] Sending {'id': 4, 'type': 'result', 'success': False, 'error': {'code': 'unknown_error', 'message': ', SetValueFailed: '}}
```
Seems to return null

Reply:
When I pass `err` directly, I get a serialization error:

```python
DEBUG    homeassistant.components.websocket_api.http.connection:http.py:76 [4566736176] Sending {'id': 2, 'type': 'result', 'success': False, 'error': {'code': 'not_supported', 'message': InvalidNewValue()}}
ERROR    homeassistant.components.websocket_api.messages:messages.py:82 Unable to serialize to JSON. Bad data found at $.error.message=(<class 'zwave_js_server.exceptions.InvalidNewValue'>
```

and when I pass `str(err)`, I get nothing:

```python
DEBUG    homeassistant.components.websocket_api.http.connection:http.py:76 [4449430688] Sending {'id': 4, 'type': 'result', 'success': False, 'error': {'code': 'unknown_error', 'message': ''}}
```

Reply:
If the exception doesn't have an argument the magic str method with return empty string.
```
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: f"{Exception()}"
Out[1]: ''

In [2]: f"{Exception('An exception argument.')}"
Out[2]: 'An exception argument.'
```

Reply:
I see, thanks, that explains the issue.